connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/admin/LoggingResource.java

```diff
@@ -0,0 +1,200 @@
+     * Adjust level of a named logger. if name corresponds to an ancestor, then the log level is applied to all child loggers.
+     *
+     * @param namedLogger name of the logger
+     * @param levelMap a map that is expected to contain one key 'level', and a value that is one of the log4j levels:
+     *                 DEBUG, ERROR, FATAL, INFO, TRACE, WARN
+     * @return names of loggers whose levels were modified
+     */
+    @PUT
+    @Path("/{logger}")
+    @SuppressWarnings("unchecked")
+    public Response setLevel(final @PathParam("logger") String namedLogger,
+                             final Map<String, String> levelMap) {
+        String desiredLevelStr = levelMap.get("level");
+        if (desiredLevelStr == null) {
+            throw new BadRequestException("Desired 'level' parameter was not specified in request.");
+        }
+
+        Level level = Level.toLevel(desiredLevelStr.toUpperCase(Locale.ROOT), null);
+        if (level == null) {
+            throw new NotFoundException("invalid log level '" + desiredLevelStr + "'.");
+        }
+
+        List<Logger> childLoggers;
+        if (ROOT_LOGGER_NAME.equals(namedLogger)) {
+            childLoggers = Collections.list(currentLoggers());
+            childLoggers.add(rootLogger());
+        } else {
+            childLoggers = new ArrayList<>();
+            Logger ancestorLogger = lookupLogger(namedLogger);
+            Enumeration en = currentLoggers();
+            boolean present = false;
+            while (en.hasMoreElements()) {
+                Logger current = (Logger) en.nextElement();
+                if (current.getName().startsWith(namedLogger)) {
+                    childLoggers.add(current);
+                }
+                if (namedLogger.equals(current.getName())) {
+                    present = true;
+                }
+            }
+            if (!present) {
+                childLoggers.add(ancestorLogger);
+            }
+        }
+
+        List<String> modifiedLoggerNames = new ArrayList<>();
+        for (Logger logger: childLoggers) {
+            logger.setLevel(level);
+            modifiedLoggerNames.add(logger.getName());
+        }
+        Collections.sort(modifiedLoggerNames);
+
+        return Response.ok(modifiedLoggerNames).build();
+    }
+
+    protected Logger lookupLogger(String namedLogger) {
+        return LogManager.getLogger(namedLogger);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected Enumeration<Logger> currentLoggers() {
+        return LogManager.getCurrentLoggers();
+    }
+
+    protected Logger rootLogger() {
+        return LogManager.getRootLogger();
+    }
+
+    /**
+     *
+     * Map representation of a logger's effective log level.
+     *
+     * @param logger a non-null log4j logger
+     * @return a singleton map whose key is level and the value is the string representation of the logger's effective log level.
+     */
+    private static Map<String, String> effLevelToMap(Logger logger) {
```

Comment:
It's better to avoid abbreviations. Assuming "eff" means "effective", then maybe:
```suggestion
    private static Map<String, String> effectiveLevelToMap(Logger logger) {
```